src/online/unigrid.jl

```diff
@@ -43,3 +43,40 @@ function updateZ(rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractV
     Zn = deepcopy(Z)
     return updateZ!(rng, Zn, alg, X)
 end
+
+export UniformGrid
+struct UniformGrid{N, T} <: AbstractVector{T} 
+    proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}
+
+    function UniformGrid(proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}) where {N, T}
```

Comment:
I don't think you need this constructor, it should be the default one.

Reply:
Right, will change. I initially had more in there, but that only got me `invalid subtyping`.

Reply:
Ah yeah you're probably right.